### PR TITLE
DEV-4683: Updating error/warning filesnames per DAIMS 2.0

### PR DESCRIPTION
**Pull request by dpb-bah**
**High level description:**
Simply including the File letters to the error warning reports

**Technical details:**
N/A

**Link to JIRA Ticket:**
[DEV-4683](https://federal-spending-transparency.atlassian.net/browse/DEV-4683)

The following are ALL required for the PR to be merged:
- [ ] Backend review completed
- [x] Unit & integration tests updated with relevant test cases
- Frontend impact assessment completed
- [x] Documentation Updated
- [x] Final format approved

### CODE REVIEW

File: dataactcore/utils/report.py [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
     """Format the csv file name for the requested file.
     @todo: unify these file names"""
     if cross_type:
         report_type_str = 'warning_' if warning else ''
-        return "submission_{}_cross_{}{}_{}.csv".format(submission_id, report_type_str, file_type, cross_type)
+        file_letters = ''
+        if file_type in FILE_TYPE_DICT_NAME_LETTER and cross_type in FILE_TYPE_DICT_NAME_LETTER:
+            file_letters = 'File_{}_to_{}_'.format(FILE_TYPE_DICT_NAME_LETTER[file_type],
+                                                   FILE_TYPE_DICT_NAME_LETTER[cross_type])
+        return "submission{}_crossfile_{}{}{}_{}.csv".format(submission_id, report_type_str, file_letters, file_type,
```
Comment (alburde1) [inline dataactcore/utils/report.py:16]: We seem to have lost an `_` here after `submission`.

File: dataactcore/utils/report.py [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
+                                                             cross_type)
     else:
-        report_type_str = 'warning' if warning else 'error'
-        return "submission_{}_{}_{}_report.csv".format(submission_id, file_type, report_type_str)
+        report_type_str = 'warning_' if warning else 'error_'
+        file_letters = ''
+        if file_type in FILE_TYPE_DICT_NAME_LETTER:
+            file_letters = 'File_{}_'.format(FILE_TYPE_DICT_NAME_LETTER[file_type])
+        return "submission{}_{}{}_{}report.csv".format(submission_id, file_letters, file_type, report_type_str)
```
Comment (alburde1) [inline dataactcore/utils/report.py:23]: We have also lost an underscore here after `submission`

File: dataactcore/utils/report.py [hunk 3]
```diff
@@ -1,18 +1,26 @@
...
 
 
 def report_file_name(submission_id, warning, file_type, cross_type=None):
     """Format the csv file name for the requested file.
     @todo: unify these file names"""
     if cross_type:
         report_type_str = 'warning_' if warning else ''
-        return "submission_{}_cross_{}{}_{}.csv".format(submission_id, report_type_str, file_type, cross_type)
+        file_letters = ''
+        if file_type in FILE_TYPE_DICT_NAME_LETTER and cross_type in FILE_TYPE_DICT_NAME_LETTER:
```
Comment (alburde1) [inline dataactcore/utils/report.py:13]: Why are these `if` statements? In what situation would we be passing a file type that's not in our list?

File: dataactcore/utils/report.py (hunk 1 above)
Comment (dpb-bah) [reply dataactcore/utils/report.py:16]: That's part of said format.
* `submission_#####award_financial_warning_report.csv `
`submission#####_File_C_award_financial_warning_report.csv`
* `submission_#####cross_warning_award_financial_award.csv`
`submission#####_crossfile_warning_File_C_to_D2_award_financial_award.csv`

File: dataactcore/utils/report.py (hunk 2 above)
Comment (dpb-bah) [reply dataactcore/utils/report.py:23]: See other comment

File: dataactcore/utils/report.py (hunk 1 above)
Comment (dpb-bah) [reply dataactcore/utils/report.py:16]: Turns out JIRA didn't update the issue on my end.

File: dataactcore/utils/report.py (hunk 3 above)
Comment (dpb-bah) [reply dataactcore/utils/report.py:13]: There's a test that does this. We can update the test? Or handle cases like the test.
https://github.com/fedspendingtransparency/data-act-broker-backend/blob/8302d223d1da63bd7d8a34ab7958edce90388f3a/tests/unit/dataactbroker/test_file_handler.py#L439-L450

Comment (alburde1) [reply dataactcore/utils/report.py:13]: I don't see where we could possibly have an invalid file type actually within the code. This seems like an old test if anything.

File: dataactcore/utils/report.py [hunk 4]
```diff
@@ -1,18 +1,22 @@
...
+from dataactcore.models.lookups import FILE_TYPE, FILE_TYPE_DICT_NAME_LETTER
 
 
 def report_file_name(submission_id, warning, file_type, cross_type=None):
     """Format the csv file name for the requested file.
     @todo: unify these file names"""
     if cross_type:
         report_type_str = 'warning_' if warning else ''
-        return "submission_{}_cross_{}{}_{}.csv".format(submission_id, report_type_str, file_type, cross_type)
+        file_letters = 'File_{}_to_{}_'.format(FILE_TYPE_DICT_NAME_LETTER[file_type],
```
Comment (alburde1) [inline dataactcore/utils/report.py:12]: Since there is no longer an `if` is this variable really necessary?

Comment (dpb-bah) [reply dataactcore/utils/report.py:12]: I kept it in for readability but can put it directly in the return

**Pull request merged**